langchain/llms/databricks_endpoint.py

```diff
@@ -0,0 +1,302 @@
+            ] = f"https://{host}/serving-endpoints/{endpoint_name}/invocations"
+        return values
+
+    def post(self, request: Any) -> Any:
+        # See https://docs.databricks.com/machine-learning/model-serving/score-model-serving-endpoints.html
+        wrapped_request = {"dataframe_records": [request]}
+        response = self.post_raw(wrapped_request)["predictions"]
+        # For a single-record query, the result is not a list.
+        if isinstance(response, list):
+            response = response[0]
+        return response
+
+
+class DatabricksClusterDriverProxyClient(DatabricksClientBase):
+    """An API client that talks to a Databricks cluster driver proxy app."""
+
+    host: str
+    cluster_id: str
+    cluster_driver_port: str
+
+    @root_validator(pre=True)
+    def set_api_url(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        if "api_url" not in values:
+            host = values["host"]
+            cluster_id = values["cluster_id"]
+            port = values["cluster_driver_port"]
+            values[
+                "api_url"
+            ] = f"https://{host}/driver-proxy-api/o/0/{cluster_id}/{port}"
+        return values
+
+    def post(self, request: Any) -> Any:
+        return self.post_raw(request)
+
+
+def get_repl_context() -> Any:
+    """Gets the notebook REPL context if running inside a Databricks notebook.
+    Returns None otherwise.
+    """
+    context = None
+    try:
+        from dbruntime.databricks_repl_context import get_context
+
+        context = get_context()
+    except ImportError:
+        logging.debug("Not running inside a Databricks notebook")
+    except Exception:
+        logging.debug(
+            "Error getting the REPL context. "
+            "Will not use it to provide default values.",
+            exc_info=True,
+        )
+    return context
+
+
+def get_default_host() -> str:
+    """Gets the default Databricks workspace hostname.
+    Raises an error if the hostname cannot be automatically determined.
+    """
+    host = os.getenv("DATABRICKS_HOST")
+    if not host:
+        try:
+            host = get_repl_context().browserHostName
+        except Exception:
+            pass
+    # TODO: support Databricks CLI profile
+    if host:
+        host = host.lstrip("https://").lstrip("http://").rstrip("/")
+    assert host, "host was not set and it cannot be automatically determined."
+    return host
+
+
+def get_default_api_token() -> str:
+    """Gets the default Databricks personal access token.
+    Raises an error if the token cannot be automatically determined.
+    """
+    api_token = os.getenv("DATABRICKS_API_TOKEN")
+    if not api_token:
+        try:
+            api_token = get_repl_context().apiToken
+        except Exception:
+            pass
+    # TODO: support Databricks CLI profile
+    assert api_token, "api_token was not set and it cannot be automatically determined."
+    return api_token
+
+
+class DatabricksEndpoint(LLM):
```

Comment:
will defer to you on this one, but could we not just call it Databricks? do we expect other databricks llm wrappers in future

Reply:
The naming follows the existing `llms.SageMakerEndpoint`. The subtle difference is that users need to deploy a model to an endpoint on their own first (similar to SageMaker) before wrapping it in LangChain. We currently do not provide pre-canned models. But I'm happy to do the rename if you feel `Databricks` is more consistent with others, similar to `Anyscale`.